ovos_skills_manager/skill_entry.py

```
@@ -28,9 +28,14 @@ def uuid(self):
         # a unique identifier
         # github_repo.github_author , case insensitive
         # should be guaranteed to be unique
-        author = self.skill_author.lower()
-        repo = self.skill_folder.lower()
-        return repo + "." + author
+        try:
+            author = self.skill_author.lower()
+            repo = self.skill_folder.lower()
+            return repo + "." + author
+        except Exception as e:
+            LOG.error(e)
+            LOG.error(f"author={self.skill_author} folder={self.skill_folder}")
```

On the other hand, are these member variables guaranteed to exist? I don't see them being initialized anywhere.

---

They are properties, could raise the null exception in those property methods, but I wasn't sure if we always want to fail on null author or folder (wouldn't matter in search for example)

---

I was thinking about the possibility of the `LOG` call raising an exception of its own:

```python3
>>> foo = SkillEntry({})
>>> print(f"{foo.skill_author}") # OK
None
>>> print(f"{foo.skill_folder}") # Here comes the pain
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/chance/hivemind-dev/ovos_skill_manager/ovos_skills_manager/skill_entry.py", line 114, in skill_folder
    return self.json.get("foldername") or self.url.split("/")[-1]
AttributeError: 'NoneType' object has no attribute 'split'
```

Incidentally, it also breaks `SkillEntry.__repr__`

---

Github auto-loaded all new comments except this one... I changed to raise an exception on skill_author not defined. If `repr` and `uuid` rely on non-null values I think it's fair to fail when either is null..

---

It probably is, but it's also truthful to print an empty SkillEntry object.

---

Oh, I see now. I definitely miscommunicated my concern.

I wasn't worried about the `skill_author` case succeeding, but rather about the `skill_folder` case failing. If these `LOG` calls raise exceptions, we're catching exceptions for the purpose of logging the occurrence and raising a specific error. If the values in question are null, I don't think the logger should raise an exception. I think it should log null and move onto the `SkillEntryError` at the end of the `except` block (not pictured above)

---

Oh, I see what you mean here; I've updated skill_folder to always return a string

---

Now the author property is doing it ;)

---

whoops.. Fixed that one now. I'll add some SkillEntry tests...